### DISCUSSION TITLE
Respond to an Inquiry via a webhook

### DISCUSSION THREAD

**Discussion by lexiismadd** [StackStorm/st2#6194] [Workflows]
Is there any way at the moment to respond to an Inquiry via a webhook.

For example, the workflow has the core.ask action in it to pause the workflow to confirm some data.  A generated URL is sent to someone via email for them to click the link to confirm the data, and when they click the link, it "responds" to the inquiry, triggering the workflow to continue..

**Comment by nzlosh** [comment]
The short answer is nothing like this exists.  It would be possible to build a solution like you've described using the API https://api.stackstorm.com/api/v1/inquiries/#/inquiries_controller.put and a custom pack to implement email processing for enquiry response validation.  I saw the email pack https://github.com/StackStorm-Exchange/stackstorm-email provides a sensor that monitors mailboxes to generate triggers based on email content which could help with the implementation.  You would also need to consider the security implications of reacting to unauthenticated email messages.

**Comment by lexiismadd** [comment]
Hi, thats not quite what I was trying to do..  What I was hoping to do is to be able to automate the Inquiry response by allowing the Inquiry response data to be sent to a webhook, rather than having users log into the platform.

For example:
1. Workflow A is triggered
2. Workflow A sends an email to a user with a URL in the body asking them to approve something.  This URL points to Webhook B
3. Workflow A reaches the core.ask point, and the workflow pauses while it waits for a response to the inquiry.
4. The end user receives the email, clicks the link in the email, which is pointing to Webhook B
5. Webhook B is triggered which in turn triggers Workflow C
6. Workflow C provides the approval to Workflow A's inquiry
8. Workflow A resumes as its received its Inquiry response

**Comment by nzlosh** [comment]
The problem with your use case is webhooks require a valid user token or api key to be called.  Which would require the user to authenticate.  You could write your own webhook app that doesn't require authentication, but then you'd have an opening in your installation for anyone to call the webhook and respond to an enquiry.

**Comment by nzlosh** [comment]
Another problem with your description is that the webhook payload for an enquiry response requires a specifically formatted JSON data structure.  This means a simple URL via email will not be sufficient to expose the enquiry parameters to respond to.  You'd need to host a landing page for users to have a UI to fill in the data correctly so it can be serialised and sent to the webhook.
